src/credentials/CHIPCertFromX509.cpp

```diff
@@ -670,6 +670,170 @@ static CHIP_ERROR ConvertCertificate(ASN1Reader & reader, TLVWriter & writer)
     return err;
 }
 
+static CHIP_ERROR ExtractPubkey(ASN1Reader & reader, Crypto::P256PublicKey & pubkey)
```

Comment:
The proposed implementation is quite complicated. The alternative is:
1. Convert X509 cert to CHIP cert
2. Load CHIP cert into ChipCertSet
3. Now you can get all certificate elements from the ChipCertificateData

Reply:
This implementation is proposed for it to work on both CHIP and non-CHIP certificates, like attestation certificates which do not need to be in CHIP format.

Reply:
Commissioner code would need to, right?  Commissionnee would not.